### ISSUE TITLE
More powerful X buttons on upload form

### ISSUE THREAD

**Issue by jidanni** [MarcusWolschon/osmeditor4android#1756]
![Screenshot_20220905-214235_Vespucci.jpg](https://user-images.githubusercontent.com/1230959/188529400-2736dcee-8dd1-4558-9167-7c10cb23b62c.jpg)
Here there is an x button next to the word "source." If we press it that St Catherine's link will go away. That is good but it's going to just come back the next time. So there needs to be two different kinds of X buttons. One to get rid of it now, another to get rid of it forever. Otherwise it'll be very easy to have it come back later just when we're not aware.. Same goes for the  comment  X above it. 

By the way at the bottom there is the word "no". Wouldn't it be more familiar to replace it with the word "cancel"?

**Comment by HolgerJeromin** [comment]
Both text fields should have a value.
So showing the last valid enty is good IMO.

**Comment by HolgerJeromin** [comment]
> By the way at the bottom there is the word "no". Wouldn't it be more familiar to replace it with the word "cancel"?

If changed this should be "close" because a text change is remembered.

**Comment by simonpoole** [comment]
At the time "No" was chosen because the semantics are clear and not easily misunderstood, it's "Upload changes" -> "No" As @HolgerJeromin  points out the last valid comment and source string gets displayed, an empty string isn't one. Unused values will eventually get removed automatically.

**Comment by jidanni** [comment]
Well I had to use a dot(.), as the program new that I was doing mischief if I just used a space bar space.
(But the iD editor doesn't remind us to give a source.
And the "every door" editor doesn't even ask us for a changeset comment.)

**Comment by simonpoole** [comment]
> (But the iD editor doesn't remind us to give a source.
> And the "every door" editor doesn't even ask us for a changeset comment.)

Quoting bad and asocial behaviour by other editors is not exactly a good way to convince me to change something.
